src/ducks/auth.ts

```diff
@@ -336,29 +947,253 @@ export const useAuthenticationStore = create<AuthStore>()((set) => ({
+   *
+   * @returns {Promise<AuthStatus>} The current authentication status
+   */
+  getAuthStatus: async () => {
+    const authStatus = await getAuthStatus();
+    set({ authStatus });
+
+    // If the hash key is expired, navigate to lock screen
+    if (authStatus === AUTH_STATUS.HASH_KEY_EXPIRED) {
+      get().navigateToLockScreen();
+    }
+
+    return authStatus;
+  },
+
+  /**
+   * Navigates to the lock screen
+   *
+   * Used when authentication expires or user needs to re-authenticate
+   */
+  navigateToLockScreen: () => {
+    const { navigationRef } = get();
+    if (navigationRef && navigationRef.isReady()) {
+      // Check if we're already on the lock screen to prevent navigation loops
+      const currentRoute = navigationRef.getCurrentRoute();
+      if (currentRoute?.name === ROOT_NAVIGATOR_ROUTES.LOCK_SCREEN) {
+        // Already on lock screen, don't navigate again
+        return;
+      }
+
+      // Use resetRoot instead of navigate to avoid warnings with conditional navigators
+      navigationRef.resetRoot({
+        index: 0,
+        routes: [{ name: ROOT_NAVIGATOR_ROUTES.LOCK_SCREEN }],
+      });
+    }
+  },
+
+  /**
+   * Fetches the active account data
+   *
+   * Checks auth status first and redirects to lock screen if hash key is expired
+   *
+   * @returns {Promise<ActiveAccount | null>} The active account or null if not found
+   */
+  fetchActiveAccount: async () => {
+    set({ isLoadingAccount: true, accountError: null });
+
+    try {
+      // Check auth status first
+      const authStatus = await getAuthStatus();
+
+      if (authStatus === AUTH_STATUS.HASH_KEY_EXPIRED) {
+        set({ authStatus: AUTH_STATUS.HASH_KEY_EXPIRED });
+
+        // Navigate to lock screen
+        get().navigateToLockScreen();
+
+        set({ isLoadingAccount: false });
+        return null;
+      }
+
+      const activeAccount = await getActiveAccount();
+      set({ account: activeAccount, isLoadingAccount: false });
+      return activeAccount;
+    } catch (error) {
+      const errorMessage =
+        error instanceof Error ? error.message : String(error);
+      set({ accountError: errorMessage, isLoadingAccount: false });
+      return null;
+    }
+  },
+
+  /**
+   * Refreshes the active account data
+   *
+   * @returns {Promise<ActiveAccount | null>} The active account or null if not found
+   */
+  refreshActiveAccount: () => get().fetchActiveAccount(),
+
+  /**
+   * Sets the navigation reference for navigation actions
+   *
+   * @param {NavigationContainerRef<RootStackParamList>} ref - The navigation reference
+   */
+  setNavigationRef: (ref) => {
+    set({ navigationRef: ref });
+  },
+
+  /**
+   * Wipes all user data for debugging purposes
+   *
+   * Removes all keys from key manager, clears all stored data,
+   * and resets the store state to initial.
+   *
+   * @returns {Promise<boolean>} True if successful, false otherwise
+   */
+  wipeAllDataForDebug: async () => {
```

Comment:
could we guard this function with a `if (!__DEV__) return;` just in case to be future proof?